### ISSUE TITLE
More useful error message for Integer macros when failing to require module

### ISSUE THREAD

**Issue by cpjolicoeur** [elixir-lang/elixir#5152]
### Environment

* Elixir version (elixir -v): 1.3.2
* Operating system: OS X 10.11.4

### Current behavior

When failing to `require Integer` in a module, using the `Integer.is_even/1` or `Integer.is_odd/1` macros produce a non-useful error message. 

```elixir
** (UndefinedFunctionError) function Integer.is_even/1 is undefined or private. Did you mean one of:

      * is_even/1

    (elixir) Integer.is_even(5)
```

### Expected behavior

Not sure exactly what message, but I'd expect a more generally useful error message in this case.  The compiler is aware of the macro definition and suggests using it, but gives a message that the function is undefined or private.

**Comment by josevalim** [comment]
Thank you for the report!

Technically speaking, there is no function `Integer.is_even/1` but a macro named `Integer.is_even/1`. We can likely make the error message better but I am also afraid of making it more verbose for the most common cases (in contrast to the one above is rather uncommon).

**Comment by cpjolicoeur** [comment]
Agreed.  That's why I said I'm not sure what message would even provide more value as it is a pretty specific situation.  Just wanted to file it in case there were other ideas.  Close if it's a non-starter, of course.

Thanks.

**Comment by lexmag** [comment]
What if we check if the mfarity in `%UndefinedFunctionError{}` matches the one in the exports, I think we can consider this as macro call without `require`. 🤔

**Comment by whatyouhide** [comment]
Agreed with @lexmag, that could work. I could try to put something together if we think it's a good approach. My 2c, this is a terrible error message because we're saying "`x` is undefined, maybe you meant `x`" which is a kind of error messages very frustrating to try and fix 😄

**Comment by josevalim** [comment]
Sounds good!

**Comment by whatyouhide** [comment]
Closed by #5156.
